pkg/spec/util.go

```diff
@@ -0,0 +1,215 @@
+		}
+	}
+	return embeddedStructs, nil
+}
+
+// This function returns an array of all the JSON tags used for unmarshalling
+// (which are the first ones specified) which have a JSON tag "conflicting", in
+// all of the input structs.
+// It's made sure that only the tags in "package spec" are checked and all other
+// packages are ignored. This is done because we are expecting the structs to
+// be handled, i.e. marked as "conflicting" only in spec.go, and we do not want
+// to populate the list with JSON tags from some other package which had the
+// "conflicting" JSON tag for some other reason
+func getMarkedAsConflictingJSONUnmarshalTags(inputStructs []interface{}) ([]string, error) {
+	var blacklistedTags []string
+	for _, inputStruct := range inputStructs {
+
+		// Checking if input is pointer to struct
+		if err := checkTypePointerToStruct(inputStruct); err != nil {
+			return nil, errors.Wrap(err, "Input parameter type mismatch")
+		}
+
+		val := reflect.ValueOf(inputStruct).Elem()
+
+		// Proceeding only if the struct belongs to the kedge's spec package
+		if val.Type().PkgPath() == kedgeSpecPackagePath {
+			for i := 0; i < val.NumField(); i++ {
+				field := val.Type().Field(i)
+
+				fieldJSONTags := strings.Split(field.Tag.Get("json"), ",")
+
+				// Adding fields that have "conflicting" as a JSON tag to the
+				// array. We need not check the first tag that is used for
+				// unmarshalling.
+				for _, tag := range fieldJSONTags[1:] {
+					if tag == "conflicting" {
+						blacklistedTags = append(blacklistedTags, fieldJSONTags[0])
+					}
+				}
+			}
+		}
+	}
+	return blacklistedTags, nil
+}
+
+// This function checks if the input parameter has the pointer to a struct as
+// its underlying type.
+// Returns "nil" if true, or an error if false
+func checkTypePointerToStruct(input interface{}) error {
+	if reflect.Indirect(reflect.ValueOf(input)).Kind().String() != "struct" {
+		return errors.New(fmt.Sprintf("Expected pointer to struct, got %v", reflect.ValueOf(input).Kind().String()))
+	}
+	return nil
+}
+
+// This function takes in pointer to a struct as input, and returns a map of
+// conflicting JSON tags used for unmarshaling.
+// All of the embedded structs are taken into account, and the fields with a
+// JSON tag "conflicting" are assumed to be handled.
+// The returned map contains the JSON tags as the keys, and the values are an
+// array of struct names which contain those fields, without being handled.
+func findConflictingJSONTags(inputStruct interface{}) (map[string][]string, error) {
+
+	// We need to check that no JSON tag is specified more than once at the
+	// top level of the struct.
+	// This means that the struct fields, and the JSON tags of all the
+	// fields of all the embedded structs need to be unique.
+	// We accomplish this over a 4 step process.
+
+	// Checking if input is pointer to struct
+	if err := checkTypePointerToStruct(inputStruct); err != nil {
```

Comment:
Not sure :( This is not making it future proof, it's making every function complete. Maybe look at it like making every input parameter type safe. Since we are taking an interface{} as an input which can take literally any type, we _must_ make sure that the type is what we want.